src/BTCUtils.sol

```diff
@@ -581,13 +581,17 @@ library BTCUtils {
             return hex"";
         }
 
-        if (uint8(_output[_at + 1]) == 0) {
+        // version opcode is 0x00 for p2wpkh and p2wsh, 0x51 for p2tr
+        uint256 _version = uint8(_output[_at + 1]);
+        if (_version == 0x00 || _version == 0x51) {
```

Comment:
Wondering if there could ever been an issue when we mistake the address for p2tr when we expect it to be p2wsh for example. Is it worth also exposing the version? I'm just thinking for Gateway if the user gives a taproot address we won't be able to check it

Reply:
Well the validation logic should fail if we find a P2TR address when the desired output address is P2WSH but with this we would bury that

Reply:
Actually nevermind the same is true of the other address types